operator/Dockerfile

```diff
@@ -1,73 +1,15 @@
-
-FROM quay.io/stackrox-io/apollo-ci:${BUILD_IMAGE_VERSION} as builder
-
-# Build the manager binary
-ARG roxpath
-
-WORKDIR ${roxpath}/
-ENV GOPATH=/workspace
-
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
-
-# Cache deps before building and copying source so that we don't need to re-download as much and so that source changes
-# don't invalidate our downloaded layer.
-# We're not using `go mod tidy` here becuase go mod tidy needs to examine _source code_ in order to find unused modules.
-# There's no source code in this docker layer yet and so `go mod tidy` would empty go.mod and go.sum which is not what
-# we want. If we're to COPY source before running `go mod tidy`, local docker build times would go up because any code
-# change will invalidate docker layers and will cause modules redownload (during `tidy`). Therefore we use
-# `go mod download` as a compromise for shorter local build times due to docker layer caching. There are CI checks
-# ensuring that there will not be unused modules for the repo overall.
-# Note that `go mod download` can donwload more stuff than `go mod tidy` https://github.com/golang/go/issues/43994
-# However it does _not_ seem to resolve packages _incorrectly_ and so should be safe especially given that downloaded
-# packages are only used during build and later this docker layer is discarded (only resulting binary goes in the final
-# image).
-# Retry as the proxy can be unavailable at times.
-ENV GOPROXY=https://proxy.golang.org|https://goproxy.io|direct
-RUN go mod download || go mod download || go mod download
-
-# Copy operator source
-COPY operator/ operator/
-
-# Copy common packages from repo root
-COPY pkg/ pkg/
-COPY image/ image/
-
-# Copy generated files from repo root.
-# Not generating them during this docker build because they anyway need to be generated by the orchestrating makefiles.
-COPY generated/ generated/
-
-# Copy scripts/go-build.sh and dependencies.
-COPY scripts/ scripts/
-COPY operator/build/status.sh status.sh
-
-# This creates a git repo in workdir so that `git grep` command in build.sh can succeed and actually find //XDef-s.
-RUN git init && git add .
-
-# We've been historically building operator without CGO both upstream and downstream. The apollo-ci image has that
-# enabled by default, therefore we disable CGO here to keep doing the thing we used to.
-ENV CGO_ENABLED=0
-
-# Build the operator binary.
-RUN GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) scripts/go-build.sh operator
-# Copy the operator binary to a location from where it will be taken into the final image.
-RUN cp -a bin/$(go env GOOS)_$(go env GOARCH)/operator stackrox-operator
-
 FROM registry.access.redhat.com/ubi8-micro:latest
 
-ARG roxpath
-
 ARG ROX_IMAGE_FLAVOR
 ENV ROX_IMAGE_FLAVOR=${ROX_IMAGE_FLAVOR}
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
 
-COPY --from=builder ${roxpath}/stackrox-operator /usr/local/bin/
+COPY operator/bin/$TARGETPLATFORM/manager /usr/local/bin/
```

Comment:
In general I'm ok with that approach as we have it on other images as well but some time ago we discussed it and agreed it's good to have dockerized builds for downstream

- https://github.com/stackrox/stackrox/pull/4301
- [slack 🧵 ](https://redhat-internal.slack.com/archives/C03D78P4RFF/p1673264810619489?thread_ts=1671727576.312279&cid=C03D78P4RFF)